aten/src/ATen/native/native_functions.yaml

```diff
@@ -6355,12 +6363,14 @@
     CUDA: foreach_tensor_div_scalar_kernel_cuda_
 
 - func: _foreach_add.List(Tensor[] tensors1, Tensor[] tensors2, *, Scalar alpha=1) -> Tensor[]
+  device_guard: False
```

Comment:
foreach tensor operators will check whether all tensors are in the same device (through `can_use_fast_route`) 

https://github.com/pytorch/pytorch/blob/8243ba72051a472a89f46a4d2c50837c14ee23fa/aten/src/ATen/native/cuda/ForeachUnaryOp.cu#L132-L148

The `slow` variant essentially redispatch the operator to each individual tensor. Thus we don't need device guard at foreach operator level:
https://github.com/pytorch/pytorch/blob/8243ba72051a472a89f46a4d2c50837c14ee23fa/aten/src/ATen/native/ForeachOpsKernels.cpp#L28-L46

We might want to obtain device guard when running through the fast path?